app/queries/coaches_resolver.rb

```diff
@@ -0,0 +1,20 @@
+class CoachesResolver < ApplicationQuery
+  include AuthorizeCoach
+
+  property :coach_ids
+  property :course_id
+
+  def coaches
+    if coach_ids.nil?
+      course.faculty
+    else
+      coach_ids.empty? ? Faculty.none : course.faculty.where(id: coach_ids)
```

Comment:
This ternary doesn't serve any purpose. You can make the query with an empty array, and it'll return an empty array. It won't even run a SQL query.

```suggestion
      course.faculty.where(id: coach_ids)
```